File: csharp/Behavioral/19-State/State/InProgressState.cs
namespace State;

public class InProgressState : BaseJobState
{
    public override string Name => "InProgress";

    public override void Pause(JobCard job, string reason)
    {
        job.PauseReason = reason;
        job.TransitionTo(new PausedState());
        Console.WriteLine($"  ⏸️  Paused — {reason}");
    }

    public override void Complete(JobCard job)
    {
        job.CompletedAt = DateTime.Now;
        job.TransitionTo(new PendingReviewState());
        Console.WriteLine("  ✔️  Work complete — awaiting review");
    }
}

File: typescript/Behavioral/19-State/State/InProgressState.ts
import { BaseJobState } from "./BaseJobState";
import { PausedState } from "./PausedState";
import { PendingReviewState } from "./PendingReviewState";
import type { JobCard } from "../Context/JobCard";

export class InProgressState extends BaseJobState {
  readonly name = "InProgress";

  override pause(job: JobCard, reason: string): void {
    job.pauseReason = reason;
    job.transitionTo(new PausedState());
    console.log(`  ⏸️  Paused — ${reason}`);
  }

  override complete(job: JobCard): void {
    job.completedAt = new Date();
    job.transitionTo(new PendingReviewState());
    console.log("  ✔️  Work complete — awaiting review");
  }
}
